example_deploy.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import re
+import shlex
+import subprocess
+import sys
+from pathlib import Path
+from typing import NamedTuple, Optional
+
+from example_utils import get_examples, ExampleType
+
+
+class DeployError(NamedTuple):
+    stdout: str
+    stderr: str
+    code: int
+
+
+def deploy(
+    deployable: bool, module_with_stub: Path, dry_run: bool, filter_pttrn: Optional[str]
+) -> Optional[DeployError]:
+    if filter_pttrn and not re.match(filter_pttrn, module_with_stub.name):
+        return None
+
+    if not deployable:
+        print(f"⏩ skipping: '{module_with_stub.name}' is not marked for deploy")
+        return None
+
+    deploy_command = f"modal app deploy {module_with_stub.name}"
+    if dry_run:
+        print(f"🌵  dry-run: '{module_with_stub.name}' would have deployed")
+    else:
+        print(f"⛴ deploying: '{module_with_stub.name}' ...")
+        r = subprocess.run(shlex.split(deploy_command), cwd=module_with_stub.parent, capture_output=True)
```

Comment:
we should figure out a way to do this not through a subprocess but by importing modal directly and calling some function – seems useful for CI type stuff in general

Reply:
Yeh I did think of that. I kept it simple for now, but we should adopt an API based approach not shell out.